client/landing/stepper/declarative-flow/internals/steps-repository/use-my-domain/index.tsx

```diff
@@ -0,0 +1,109 @@
+import { useMyDomainInputMode as inputMode } from 'calypso/components/domains/connect-domain-step/constants';
+import UseMyDomainComponent from 'calypso/components/domains/use-my-domain';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { ONBOARD_STORE } from 'calypso/landing/stepper/stores';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { domainMapping, domainTransfer } from 'calypso/lib/cart-values/cart-items';
+import CalypsoShoppingCartProvider from 'calypso/my-sites/checkout/calypso-shopping-cart-provider';
+import type { Step } from '../../types';
+
+import './style.scss';
+
+const UseMyDomain: Step = function UseMyDomain( { navigation, flow } ) {
+	const { setHideFreePlan, setDomainCartItem } = useDispatch( ONBOARD_STORE );
+	const { goNext, goBack, submit } = navigation;
+	const { __ } = useI18n();
+	const isStartWritingFlow = 'start-writing' === flow;
+	const getDefaultStepContent = () => <h1>Choose a domain step</h1>;
+
+	const handleOnTransfer = async ( domain: string, authCode: string ) => {
+		const domainCartItem = domainTransfer( {
+			domain,
+			extra: {
+				auth_code: authCode,
+				signup: true,
+			},
+		} );
+		setHideFreePlan( true );
+		setDomainCartItem( domainCartItem );
+
+		submit?.();
+	};
+
+	const handleOnConnect = async ( domain: string ) => {
+		const domainCartItem = domainMapping( { domain } );
+		setHideFreePlan( true );
+		setDomainCartItem( domainCartItem );
+
+		submit?.();
+	};
+
+	const getInitialQuery = function () {
+		return String( getQueryArg( window.location.search, 'lastQuery' ) ?? '' );
+	};
+
+	const getStartWritingFlowStepContent = () => {
+		return (
+			<CalypsoShoppingCartProvider>
+				<UseMyDomainComponent
+					analyticsSection="signup"
+					basePath=""
+					initialQuery={ getInitialQuery() }
+					initialMode={ inputMode.domainInput }
+					isSignupStep={ true }
+					showHeader={ false }
+					hideHeader={ true }
+					onTransfer={ handleOnTransfer }
+					onConnect={ ( { domain } ) => handleOnConnect( domain ) }
+				/>
+			</CalypsoShoppingCartProvider>
+		);
+	};
+
+	const getStepContent = () => {
+		switch ( flow ) {
+			case 'start-writing':
+				return getStartWritingFlowStepContent();
+			default:
+				return getDefaultStepContent();
+		}
+	};
+
+	const getFormattedHeader = () => {
+		if ( isStartWritingFlow ) {
+			return (
+				<FormattedHeader
+					id="choose-a-domain-writer-header"
+					headerText={ __( 'Use a domain I own' ) }
+					align="center"
+				/>
+			);
+		}
+	};
+
+	return (
+		<>
+			<QueryProductsList />
+			<StepContainer
+				stepName="useMyDomain"
+				shouldHideNavButtons={ isStartWritingFlow }
+				goBack={ goBack }
+				goNext={ goNext }
+				isHorizontalLayout={ false }
+				isWideLayout={ true }
+				isLargeSkipLayout={ false }
+				stepContent={ getStepContent() }
+				recordTracksEvent={ recordTracksEvent }
+				formattedHeader={ getFormattedHeader() }
```

Comment:
I'm wondering if we passed `formattedHeader={ undefined }` here, if we could simplify a couple things by using the header in `UseMyDomainComponent`. Might be able to touch one less file and create one less function. How do you see it?

Reply:
Discussed here p1683721966816739-slack-CKZHG0QCR and implemented. ☺️